modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/security/permission/resource/AssetListEntryPortletResourcePermission.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.asset.list.internal.security.permission.resource;
+
+import com.liferay.asset.list.constants.AssetListConstants;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Jürgen Kappler
+ */
+@Component(
+	immediate = true,
+	property = "resource.name=" + AssetListConstants.RESOURCE_NAME,
+	service = PortletResourcePermission.class
+)
+public class AssetListEntryPortletResourcePermission
+	implements PortletResourcePermission {
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker, Group group, String actionId)
+		throws PrincipalException {
+
+		if (!contains(permissionChecker, null, actionId)) {
```

Comment:
Why are you using null here for a GroupedModel?  I think the only ones of these that should be passing null in is models without groupIds.  Shouldn't you just use the factory with the built in behavior?